### DISCUSSION TITLE
Subtype mapping with `Map` value

### DISCUSSION THREAD

**Discussion by patrickdronk** [instancio/instancio#727] [Q&A] [answered]
Hey, me again 😄 

So I have a class which has two maps:

```java
public class MyClass {
  private Map<String,  AbstractClass> someName;
  private Map<String, AbstractClass> someOtherName
}
```
```java
public abstract class AbstractClass{

}
```
```java
public class SomeClass extends AbstractClass {

}
```
```java
public class SomeOtherClass extends AbstractClass {

}
```

I want my map to be created by Instancio. 
And for `someName` I want it to subtype to SomeClass
For `someOtherName` I want it to subtype to SomeOtherClass. 

I have tried some things, but haven't been able to get something I liked. 

Thanks you for your time :)

**Comment by patrickdronk** [comment, marked as answer]
I was a bit to soon with the question. For people with a similar issue; read the part of scopes. 

```java
var someNameScope = field(MyClass::getSomeName).toScope()
var someOtherNameScope = field(MyClass::getSomeOtherName).toScope()

var obj = Instancio.of(MyClass:.class)
.subtype(all(AbstractClass.class).within(someNameScope), SomeClass)
.subtype(all(AbstractClass.class).within(someOtherNameScope), SomeOtherClass)
```

**Reply by armandino** [reply]
@patrickdronk thank you for providing the answer! :)

Here's a link to the docs for anyone looking for more information about selector scopes:

https://www.instancio.org/user-guide/#selector-scopes
